tools/shared/schema_cache.py

```
@@ -0,0 +1,463 @@
+        """Remove expired schema entries from cache."""
+        expired_keys = []
+
+        for key, entry in self._cache.items():
+            if self._is_expired(entry):
+                expired_keys.append(key)
+
+        for key in expired_keys:
+            del self._cache[key]
+            logger.debug(f"Removed expired schema entry: {key}")
+
+    def _validate_schema(self, schema: Dict[str, Any]) -> bool:
```

Ruff (0.11.9)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)